lib/ansible/modules/windows/win_find.ps1

```diff
@@ -172,6 +173,31 @@ Function Assert-Pattern($info) {
     $valid_match
 }
 
+Function Assert-Contains($info) {
+    $valid_match = $false
+
+    if ($contains -ne $null) {
+        $file_contents = [System.IO.File]::ReadAllText($info.path)
```

Comment:
I don't like that it's not in the `Try{} Catch{}`.
[The list of exception of the `[System.IO.File]::ReadAllText()` method](
https://docs.microsoft.com/en-us/dotnet/api/system.io.file.readalltext?view=netframework-4.5).

And what about adding additional parameter `Encoding` with the default `UTF8Encoding`?
[The System.Text.Encoding class documentation](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding?view=netframework-4.5).

Reply:
There's not too many reasons why we should be catching all the exceptions. In devel/2.8 it will show us the proper exception in the output. Unless;

* we can handle the exception
* get more info in case of a specific one
* the exception message isn't really helpful and we can do better

then it's just noise and extra code. Historically the unhandled exception output was pretty poor and unhelpful but with https://github.com/ansible/ansible/pull/45334 it's gotten a lot better. You can see some examples of how that changed the output.

Reply:
In terms of the encoding that's a good question, we should default to having `[System.IO.File]::ReadAllText()` handle that but allow the user to specify their own encoding if they wish. Having tests for this would be a must though as these are some pretty edge cases.